File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if(n==0) return ;

        if(m==0) nums1=nums2;

        int i=m-1;
        int j=n-1;
        int pos=m+n-1;

        while(i>=0 && j>=0){
            if(nums1[i]>nums2[j]){
                nums1[pos]=nums1[i];
                i--;
            }else{
                nums1[pos]=nums2[j];
                j--;
            }
            pos--;
        }
        for(int i=j;i>=0;i--){
            nums1[pos]=nums2[i];
            pos--;
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) { 
        if(n == 0){
            return;
        }

        int x = m-1;
        int y = n-1;
        int i = 0;
        if(x < 0){
            y=0;
            for( ;i < m+n; i++){
                nums1[i] = nums2[y];
                y++;
            }
            return;
        }

        i = (m+n)-1;
        for( ; i>=0;i--){
            if(y < 0 || x < 0){
                break;
            }

            if(nums2[y]> nums1[x]){
                nums1[i] = nums2[y];
                y--;
            } else {
                nums1[i] = nums1[x];
                x--;
            }
        }
        if(y >= 0){
            for(;i>=0;i--){
                nums1[i] = nums2[y];
                y--;
            }
        }
    }
}